addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionEvents.java

```diff
@@ -0,0 +1,112 @@
+
+import java.util.Calendar;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.event.DocumentEventTypes;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.test.CapturingEventListener;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+public class TestRetentionEvents extends RetentionTestCase {
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Test
+    public void shouldNotifyEventsWhenModifyLegalHold() {
+        session.makeRecord(file.getRef());
+
+        checkLegalHoldEvents(true, "I put the legal hold", DocumentEventTypes.BEFORE_SET_LEGAL_HOLD,
+                DocumentEventTypes.AFTER_SET_LEGAL_HOLD);
+
+        checkLegalHoldEvents(false, "I remove the legal hold", DocumentEventTypes.BEFORE_REMOVE_LEGAL_HOLD,
+                DocumentEventTypes.AFTER_REMOVE_LEGAL_HOLD);
+
+    }
+
+    protected void checkLegalHoldEvents(boolean hold, String comment, String... expectedEvents) {
+        try (CapturingEventListener listener = new CapturingEventListener(expectedEvents)) {
+            session.setLegalHold(file.getRef(), hold, comment);
+            transactionalFeature.nextTransaction();
+
+            String[] eventsNames = listener.streamCapturedEvents().map(Event::getName).toArray(String[]::new);
+            assertArrayEquals(expectedEvents, eventsNames);
+            listener.streamCapturedEvents()
+                    .forEach(ev -> assertEquals("Event: " + ev.getName(), //
+                            comment, ev.getContext().getProperties().get("comment")));
+        }
+    }
+
+    @Test
+    public void shouldNotifyEventsWhenModifyRetention() {
+        session.makeRecord(file.getRef());
+        Calendar retainUntil = Calendar.getInstance();
+        retainUntil.add(Calendar.DAY_OF_MONTH, 5);
+
+        // current retention is null
+        checkRetentionEvents(retainUntil, DocumentEventTypes.BEFORE_SET_RETENTION,
+                DocumentEventTypes.AFTER_SET_RETENTION);
+
+        // extend the retention
+        retainUntil = Calendar.getInstance();
+        retainUntil.add(Calendar.DAY_OF_MONTH, 25);
+        checkRetentionEvents(retainUntil, DocumentEventTypes.BEFORE_EXTEND_RETENTION,
+                DocumentEventTypes.AFTER_EXTEND_RETENTION);
+
+        // current retention is indeterminate
+        checkRetentionEvents(RETAIN_UNTIL_INDETERMINATE, DocumentEventTypes.BEFORE_EXTEND_RETENTION,
+                DocumentEventTypes.AFTER_EXTEND_RETENTION);
+
+        // modify an indeterminate once
+        retainUntil = Calendar.getInstance();
+        retainUntil.add(Calendar.DAY_OF_MONTH, 10);
+        checkRetentionEvents(retainUntil, DocumentEventTypes.BEFORE_SET_RETENTION,
+                DocumentEventTypes.AFTER_SET_RETENTION);
+    }
+
+    protected void checkRetentionEvents(Calendar retainUntil, String... expectedEvents) {
```

Comment:
I would go for `setRetentionAndCheckEvents` because this method doesn't just check events.